src/Microsoft.AspNetCore.Routing/Patterns/RoutePatternConstraintReference.cs

```diff
@@ -0,0 +1,47 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Diagnostics;
+
+namespace Microsoft.AspNetCore.Routing.Patterns
+{
+    /// <summary>
+    /// The parsed representation of a constraint in a <see cref="RoutePattern"/> parameter.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerToString()}")]
+    public sealed class RoutePatternConstraintReference
+    {
+        internal RoutePatternConstraintReference(string rawText, string name, string content)
```

Comment:
Assign name to Name property. Could you also add a sample in a comment here so that its clear (mainly how `rawText` and `content` differ is not clear for me)

Reply:
Nit: name => parameterName

Reply:
I've been using the property/variable name `name` for parameter name. I can change it here but I will also change it in a few other places. @JamesNK @kichalla thoughts?

Also probably `text` vs `rawText`

Reply:
> I've been using the property/variable name name for parameter name. I can change it here but I will also change it in a few other places. @JamesNK @kichalla thoughts?

Sure 👍